src/solc_0.8/ReferralValidator/ReferralValidator08.sol

```diff
@@ -0,0 +1,186 @@
+/* solhint-disable not-rely-on-time, func-order */
```

Comment:
You don't have to rename the contract. But let's move it to common and name it `WithReferralValidator`
`src/solc_0.8/common/BaseWithStorage/WithReferralValidator.sol`

Reply:
fixed -> c21e258